src/librustc_trans/back/write.rs

```diff
@@ -687,8 +692,11 @@ pub fn run_passes(sess: &Session,
     let link_obj = |output_path: &Path| {
         // Running `ld -r` on a single input is kind of pointless.
         if sess.opts.cg.codegen_units == 1 {
-            fs::copy(&crate_output.with_extension("0.o"),
-                     output_path).unwrap();
+            let input_path = &crate_output.with_extension("0.o");
+            if let Err(e) = fs::copy(input_path, output_path) {
+                sess.err(&format!("could not copy {:?} to {:?}: {}",
+                                  input_path, output_path, e));
+            }
```

Comment:
This duplication was nagging me in the back of my head as well, but I found the functions different enough for duplication to be passable. I extracted graceful copy function into its own for now.